test/sqllogictest/transform/union_cancel.slt

```diff
@@ -196,25 +196,7 @@ WITH t3_with_key AS (select f1 as key, sum(f2) as nokey from t3 group by f1)
 SELECT a1.* FROM t3 AS a1 LEFT JOIN t3_with_key AS a2 ON (a1.f1 = a2.key);
```

Comment:
This query no longer tests union cancellation, so strictly speaking does not belong here, but in outer_join_simplification.slt

Reply:
I can move it. I wasn't sure if it tested it in the sense that without union cancelation it would be something horrible, though I agree that it feels more like the first example from https://github.com/MaterializeInc/materialize/issues/6987 though.